docs/reference/remap.cue

```diff
@@ -179,19 +187,31 @@ remap: #Remap & {
 			title: "Invalid argument type"
 			input: log: not_a_string: 1
 			source: """
-				upcase(.not_a_string)
+				upcase(42)
 				"""
 			raises: compiletime: """
-				error: invalid argument type
-				  ┌─ :1:1
+				error[E110]: invalid argument type
+				  ┌─ :1:8
 				  │
-				1 │ upcase(.not_a_string)
-				  │        ^^^^^^^^^^^^^
+				1 │ upcase(42)
+				  │        ^^
 				  │        │
-				  │        this expression resolves to unknown type
+				  │        this expression resolves to the exact type "integer"
 				  │        but the parameter "value" expects the exact type "string"
 				  │
-				  = see language documentation at: https://vector.dev/docs/reference/vrl/
+				  = try: ensuring an appropriate type at runtime
+				  =
+				  =     42 = string!(42)
+				  =     upcase(42)
+				  =
+				  = try: coercing to an appropriate type and specifying a default value as a fallback in case coercion fails
+				  =
+				  =     42 = to_string(42) ?? "default"
+				  =     upcase(42)
+				  =
+				  = see documentation about error handling at https://errors.vrl.dev/#handling
+				  = learn more about error code 110 at https://errors.vrl.dev/110
+				  = see language documentation at https://vrl.dev
```

Comment:
Shouldn't this also suggest an error handling solution given the latest changes?

Reply:
No, in fact, this should really provide _no_ solution at all because there is no path in which passing `42` to `upcase` can work.

The "invalid argument type" error now only shows up when there is no way the given argument can ever succeed. Meaning, if its type definition does not overlap with the accepted type definition, this error is raised.